apps/snitch_core/test/domain/order/transitions_test.exs

```diff
@@ -141,4 +148,36 @@ defmodule Snitch.Domain.Order.TransitionsTest do
       assert {:ok, %{packages: []}} = Repo.transaction(result.multi)
     end
   end
+
+  describe "associate_package" do
+    setup do
+      order = insert(:order, user: build(:user))
+
+      packages =
+        insert_list(
+          1,
+          :package,
+          order_id: order.id,
+          origin: build(:stock_location),
+          shipping_methods: build_list(1, :embedded_shipping_method),
+          shipping_category: build(:shipping_category)
+        )
+
+      [order: order, packages: packages]
+    end
+
+    test "with packages", %{order: order, packages: packages} do
```

Comment:
The setup built here does not accurately mimic the real scenario. It was made crystal clear in the [meeting](https://hackmd.io/atLBxkpvReiEwSbx5CzUTA#Live-coding) that when the transition is called, the `package.shipping_method_id` is `nil`. You can't set the `:id` like this on L170.
Moreover, what is the significance of this `3`?